lib/DDGC/Web/Controller/InstantAnswer.pm

```diff
@@ -50,11 +52,35 @@ sub index :Chained('base') :PathPart('') :Args(0) {
     # @{$c->stash->{ialist}} = $c->d->rs('InstantAnswer')->all();
 }
 
+sub _add_json_last_modified_header {
+    my ( $self, $c, $last_modified ) = @_;
+    $last_modified->set_formatter( 'DateTime::Format::HTTP' ) if $last_modified;
+    my $if_modified_since = $c->req->header('If-Modified-Since');
+
+    $if_modified_since = DateTime::Format::HTTP->parse_datetime(
+        $if_modified_since,
+    ) if $if_modified_since;;
+
+    $c->response->header(
+        'Last-Modified' => "$last_modified",
+    );
+    if ( ( $if_modified_since && $last_modified ) &&
+         DateTime->compare( $if_modified_since, $last_modified ) <= 0 ) {
+        $c->response->status('304');
+        return $c->detach;
+    }
+
+    return $c->detach if ( $c->request->method eq 'HEAD' );
```

Comment:
This sub:

- Sets `Last-Modified` response header
- Short-circuits the request if `Last-Modified` >= `If-Modified-Since` request header or request method is HEAD.